xtensa-lx/src/interrupt.rs

```diff
@@ -158,17 +156,15 @@ pub fn get_level() -> u32 {
 #[inline]
 pub fn free<F, R>(f: F) -> R
```

Comment:
Could we add a quick comment to mention that this is not safe on multicore? Maybe we should even mark it as unsafe?

Reply:
I'm fairly sure this is fine as is; this function doesn't produce a critical section or do anything with other cores, it just disables/re-enables interrupts.

Reply:
Although the original comment was incorrectly referring to being equal to critical sections